### ISSUE TITLE
[KED-2376] Fix links to make them all accessible

### ISSUE THREAD

**Issue by yetudada** [kedro-org/kedro#1281]
Some of our links are still failing basic accessibility which is fine for now but we should fix them someday. We can adjust some just as we work on pages, but should put some time aside to go through and fix them all. This change should only be done after we release the new URL structure in 0.18.0. 

This [section is taken from our style guide](https://kedro.readthedocs.io/en/stable/14_contribution/04_documentation_contributor_guidelines.html). Make hyperlink descriptions as descriptive as you can. This is a good description:

```
Learn how to [update the project pipeline](https://kedro.readthedocs.io/en/stable/03_tutorial/04_create_pipelines.html#update-the-project-pipeline)
```

This is less helpful:

```
Learn how to update the [project pipeline](https://kedro.readthedocs.io/en/stable/03_tutorial/04_create_pipelines.html#update-the-project-pipeline)
```

Don’t write this:

```
To learn how to update the project pipeline, see [here](https://kedro.readthedocs.io/en/stable/03_tutorial/04_create_pipelines.html#update-the-project-pipeline)
```

Links should be descriptive rather than single words. They don't need to be full sentences but they need to be meaningful so that anyone reading with a screenreader will be able to know what is behind a link before they follow it.
